__tests__/pet.test.js

```diff
@@ -0,0 +1,149 @@
+	});
+});
+
+describe('Keeping fed', () => {
+	const pet = new Pet('Fido');
+	test('check when feed() hunger decreases by 3, with min 0', () => {
+		pet.feed();
+		expect(pet.hunger).toBe(0);
+		pet.feed();
+		expect(pet.hunger).toBe(0);
+	});
+});
+
+describe('Check up', () => {
+	const pet = new Pet('Fido');
+	test('check new instance is feeling great', () => {
+		// on initialisation fitess is 10, and hunger is 0
+		expect(pet.checkUp()).toBe('I feel great!');
+	});
+	test('check needs walk', () => {
+		pet.fitness = 3;
+		expect(pet.checkUp()).toBe('I need a walk');
+	});
+	test('check needs feeding', () => {
+		pet.fitness = 10;
+		pet.hunger = 5;
+		expect(pet.checkUp()).toBe('I am hungry');
+	});
+	test('check needs both walking and feeding', () => {
+		pet.fitness = 2;
+		pet.hunger = 6;
+		expect(pet.checkUp()).toBe('I need walking and feeding');
+	});
+});
+
+describe('Death ☠️', () => {
+	const pet = new Pet('Fido');
+	test('check alive when born', () => {
+		expect(pet.isAlive).toBe(true);
+	});
+	test('check death at 30', () => {
+		pet.age = 30;
+		expect(pet.isAlive).toBe(false);
+	});
+	test('check death when hunger reaches 10', () => {
+		pet.age = 0;
+		pet.hunger = 10;
+		expect(pet.isAlive).toBe(false);
+	});
+	test('check death with fitness reaches 0', () => {
+		pet.age = 0;
+		pet.hunger = 0;
+		pet.fitness = 0;
+		expect(pet.isAlive).toBe(false);
+	});
+});
+
+describe('Guard Clauses', () => {
+	const pet = new Pet('Fido');
+	test('check checkUp() alive', () => {
+		expect(pet.checkUp()).toBe('I feel great!');
+	});
+	test('check checkUp() dead', () => {
+		pet.fitness = 0;
+		expect(pet.checkUp()).toBe('Your pet is no longer alive :(');
+	});
+	test('check functions throw error when dead', () => {
+		expect(() => pet.growUp()).toThrow('Your pet is no longer alive :(');
+		expect(() => pet.walk()).toThrow('Your pet is no longer alive :(');
+		expect(() => pet.feed()).toThrow('Your pet is no longer alive :(');
+	});
+});
+
+describe('Making Babies', () => {
+	const parentPet = new Pet('Big Daddy');
+	const childPet = new Pet('Junior');
+	const siblingPet = new Pet('Lil Sis');
+	const grandChildPet = new Pet('Lil Junior');
+	parentPet.adoptChild(childPet);
+	parentPet.adoptChild(siblingPet);
+	siblingPet.adoptChild(grandChildPet);
+	test('check child is adopted', () => {
+		expect(parentPet.children[0]).toBe(childPet);
+		expect(parentPet.children[0].getName()).toBe('Junior');
+		expect(parentPet.children[1].getName()).toBe('Lil Sis');
+		expect(parentPet.children[1].children[0].getName()).toBe('Lil Junior');
+		expect(siblingPet.children[0].getName()).toBe('Lil Junior');
+	});
+	test('check adoption', () => {
+		parentPet.adoptChild({ name: 'Dave', species: 'Human' });
+		expect(parentPet.children[2].name).toBe('Dave');
+	});
+	test('check baby is made', () => {
+		parentPet.haveBaby('Cutie Pie');
+		expect(parentPet.babies[0].getName()).toBe('Cutie Pie');
+		expect(parentPet.babies[0]).toBeInstanceOf(Pet);
+	});
```

Comment:
Unlike the adopt tests, you can get away with calling `getName` here, as otherwise you can't test that the created baby has the name that you expect. You can however make this code more DRY by assigning this name to a variable beforehand, e.g.
```
const babyName = "Cutie Pie";
const parent = new Pet("Big Daddy");
parent.haveBaby(babyName);

expect(parentPet.babies[0].getName()).toBe(babyName);
...
```

Reply:
As mentioned previously, I think you should use the same array for both children and babies. You should then also do these test setup inside of the test itself to avoid them interacting with each other. Atm, this would mean a lot of repeated code (e.g. you will be doing `const pet = new Pet("Fido");` a lot). In cruise ship we get over a way to avoid this code repetition using a `beforeEach`, but for now, it's better to write your code this way then to have multiple tests that interact with the same pet